### ISSUE TITLE
CA1062 false positive when switching on (obj?.Property)

### ISSUE THREAD

**Issue by jnm2** [dotnet/roslyn-analyzers#2586]
@mavasani Is there a way to find out if https://github.com/dotnet/roslyn-analyzers/pull/2584 already fixes this?

No diagnostic should be reported in this method:

```cs
public static void Foo(Task task)
{
    switch (task?.Status)
    {
        // ...

        case TaskStatus.Faulted:
            //  ↓ CA1062 In externally visible method 'void Foo(Task task)', validate parameter 'task' is non-null before using it. If appropriate, throw an ArgumentNullException when the argument is null or add a Code Contract precondition asserting non-null argument.
            _ = task.Exception;
            break;

        // ...
    }
}
```

#### Analyzer package
https://www.nuget.org/packages/Microsoft.CodeQuality.Analyzers/

#### Package Version

2.9.3

#### Diagnostic ID

https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1062-validate-arguments-of-public-methods

**Comment by jnm2** [comment]
I'm also getting this from `dotnet build` using SDK 3.0.100-preview3-010431 at the same source location:

> warning CS8602: Possible dereference of a null reference.

Maybe this is a Roslyn bug.

**Comment by jnm2** [comment]
Perhaps this is a manifestation of https://github.com/dotnet/roslyn/issues/35668?

**Comment by mavasani** [comment]
No this is a separate bug which manifests in presence of conditional access inside a conditional. I am trying to fix this one now.

**Comment by mavasani** [comment]
https://dotnet.myget.org/feed/roslyn-analyzers/package/nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/2.9.4-beta1.19317.5+45537f7f should have the fix

**Comment by jnm2** [comment]
Thank you, I appreciate it!
